tests/e2e/utils/auto-fsx-setup.py

```diff
@@ -0,0 +1,458 @@
+import argparse
+import boto3
+import subprocess
+from shutil import which
+from time import sleep
+from utils import (
+    rand_name,
+    write_json_file,
+    get_eks_client,
+    get_ec2_client,
+    get_iam_client,
+    get_fsx_client,
+    get_iam_resource,
+    kubectl_apply_kustomize,
+    kubectl_apply,
+)
+
+
+def main():
+    header()
+
+    verify_prerequisites()
+
+    setup_iam_authorization()
+    setup_fsx_driver()
+    setup_fsx_file_system()
+    setup_fsx_provisioning()
+
+    footer()
+
+
+def header():
+    print("=================================================================")
+    print("                          FSx Setup")
+
+
+def verify_prerequisites():
+    print("=================================================================")
+    print("                   Prerequisites Verification")
+    print("=================================================================")
+
+    verify_oidc_provider_prerequisite()
+    verify_eksctl_is_installed()
+    verify_kubectl_is_installed()
+
+
+def verify_oidc_provider_prerequisite():
+    print("Verifying OIDC provider...")
+
+    if is_oidc_provider_present():
+        print("OIDC provider found")
+    else:
+        raise Exception(
+            f"Prerequisite not met : No OIDC provider found for cluster '{CLUSTER_NAME}'!"
+        )
+
+
+def is_oidc_provider_present() -> bool:
+    iam_client = get_iam_client(CLUSTER_REGION)
+    oidc_providers = iam_client.list_open_id_connect_providers()[
+        "OpenIDConnectProviderList"
+    ]
+
+    if len(oidc_providers) == 0:
+        return False
+
+    for oidc_provider in oidc_providers:
+        oidc_provider_tags = iam_client.get_open_id_connect_provider(
+            OpenIDConnectProviderArn=oidc_provider["Arn"]
+        )["Tags"]
+
+        if any(
+            tag["Key"] == "alpha.eksctl.io/cluster-name"
+            and tag["Value"] == CLUSTER_NAME
+            for tag in oidc_provider_tags
+        ):
+            return True
+
+    return False
+
+
+def verify_eksctl_is_installed():
+    print("Verifying eksctl is installed...")
+
+    is_prerequisite_met = which("eksctl") is not None
+
+    if is_prerequisite_met:
+        print("eksctl found!")
+    else:
+        raise Exception(
+            "Prerequisite not met : eksctl could not be found, make sure it is installed or in your PATH!"
+        )
+
+
+def verify_kubectl_is_installed():
+    print("Verifying kubectl is installed...")
```

Comment:
I would encourage to put such generic function in utils file. 
also code for such functions can be combined in just one function called `verify_prerequisite_is_installed()` and you can check for all such pre-requisite once.